### PR TITLE
Fix #328003 score property crash

### PR DESCRIPTION
**Pull request by jonnymoo**
Resolves: https://musescore.org/en/node/328003

Following accessibility changes saving score properties causes MuseScore to crash

- [ x] I signed [CLA](https://musescore.org/en/cla)
- [ x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [ x] I made sure the code compiles on my machine
- [ x] I made sure there are no unnecessary changes in the code
- [ x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [ x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [ x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ x] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/notation/view/widgets/scoreproperties.cpp [hunk 1]
```diff
@@ -248,31 +249,31 @@ bool ScorePropertiesDialog::save()
         QLayoutItem* tagItem   = scrollAreaLayout->itemAtPosition(i, 0);
         QLayoutItem* valueItem = scrollAreaLayout->itemAtPosition(i, 1);
         if (tagItem && valueItem) {
-            QLineEdit* tag   = static_cast<QLineEdit*>(tagItem->widget());
-            QLineEdit* value = static_cast<QLineEdit*>(valueItem->widget());
 
-            QString tagText = tag->text();
+            ScorePropertiesItem item(tagItem->widget(), valueItem->widget());
```
Comment (RomanPudashkin) [inline src/notation/view/widgets/scoreproperties.cpp:253]: This class is a bit redundant. You can just add missing nullptr checks where the crash occurs

Comment (jonnymoo) [reply src/notation/view/widgets/scoreproperties.cpp:253]: Is that why the test runs have all failed except on windows? Have I added the file incorrectly?

I split it out into a new class to a) fullfill the single responsibility principle of SOLID and b) it made it much easier to unit test, but if it creates other issues then yes of course just doing the cast check in the score properties page will work. 

I’ll have a think how I can TDD that…

Comment (jonnymoo) [reply src/notation/view/widgets/scoreproperties.cpp:253]: Ps I think you are correct btw - I’ll go and simplify.

Comment (RomanPudashkin) [reply src/notation/view/widgets/scoreproperties.cpp:253]: These principles are good but in this case, your solution overcomplicates the dialog implementation. In addition, the dialog will be rewritten using the qml / c++ pair in the future (or completely replaced with something else). Anyway, thanks for your work!

Comment (jonnymoo) [reply src/notation/view/widgets/scoreproperties.cpp:253]: Simple fix as per above on way....

**Pull request closed without merging**